e2e/typescript/src/assets/fixture.ts

```diff
@@ -103,20 +117,61 @@ assertType({
 // Banned shorthand properties
 //
 
-// @ts-expect-error "padding" is banned
-assertType({ padding: 0 });
-// @ts-expect-error "borderLeft" is banned
-assertType({ borderLeft: '5px' });
+assertType({
+  // @ts-expect-error "padding" is banned
+  padding: 0,
+});
+assertType({
+  // @ts-expect-error "borderLeft" is banned
+  borderLeft: '5px',
+});
 
 // Invalid values
 //
 
-// @ts-expect-error "1" is invalid value for "overflow"
-assertType({ overflow: '1' });
-// @ts-expect-error "paddingLeft" cannot be numeric value
-assertType({ paddingLeft: 5 });
-// @ts-expect-error "0" is invalid value for "color"
-assertType({ color: 0 });
+assertType({
+  // @ts-expect-error "outline-box" is an invalid value for "box-sizing"
+  boxSizing: 'outline-box',
+});
+assertType({
+  // @ts-expect-error "1" is invalid value for "overflow"
+  overflow: '1',
+});
+assertType({
+  // @ts-expect-error "paddingLeft" cannot be numeric value
+  paddingLeft: 5,
+});
+assertType({
+  // @ts-expect-error "0" is invalid value for "color"
+  color: 0,
+});
+
+assertType({
+  // @ts-expect-error type check still fails on outline-box, not on any other line
+  boxSizing: 'outline-box',
+  zIndex: 1,
+});
+
+assertType({
+  // @ts-expect-error Object is not assignable to CSS property
+  zIndex: { color: 'red' },
+  // @ts-expect-error Object is not assignable to CSS property
+  opacity: { color: 'red' },
+});
+
+// Mixins with invalid values
+//
+
+// @ts-expect-error Object in "zIndex" is not assignable to CSS property
+assertType({
+  ...{ zIndex: { color: 'red' } },
```

Comment:
We should also add a mixin which is typed as `GriffelStyle` (similar to what we have in babel fixtures)

Reply:
added